src/main/java/org/spongepowered/common/data/SpongeDataRegistry.java

```diff
@@ -206,6 +202,11 @@ public static void finalizeRegistration() {
         }
         checkArgument(!immutableProcessorList.contains(processor), "Duplicate DataProcessor Registration!");
         immutableProcessorList.add(processor);
+        
+        SpongeSerializationService service = (SpongeSerializationService) Sponge.getGame().getServiceManager().provide(SerializationService.class).get();
+        SpongeDataBuilder<T, I> builder = new SpongeDataBuilder<>(implClass, processor);
+        service.registerBuilderAndImpl(manipulatorClass, implClass, builder);
```

Comment:
Better move Builder registration to `finalizeRegistration()`. Since multiple `DataProcessor`s may be registered for any `DataManipulator` you must wait until all are registered before creating the builder from the `DataProcessorDelegate`. Otherwise, you'd be getting errors since there can only be one builder per `DataManipulator`.